audio/decode/ad_spdif.c

```diff
@@ -419,12 +420,18 @@ static struct mp_decoder *create(struct mp_filter *parent,
 
     spdif_ctx->codec_id = mp_codec_to_av_codec_id(codec->codec);
 
+    if (spdif_ctx->codec_id == AV_CODEC_ID_NONE)
+        goto error;
+
+    spdif_ctx->avpkt = av_packet_alloc();
+    if (!spdif_ctx->avpkt)
+        goto error;
```

Comment:
alternative to error checking: `MP_HANDLE_OOM(spdif_ctx->avpkt)`
depends on whether we consider an avpkt a "small" allocation